cas-client-core/src/main/java/org/jasig/cas/client/util/CommonUtils.java

```diff
@@ -438,8 +439,12 @@ public static String getResponseFromServer(final URL constructedUrl, final HttpU
             }
 
             return builder.toString();
+        } catch (final SSLException e) {
+            LOGGER.error("SSL error getting response from host: {} : Error Message: {}", constructedUrl.getHost(), e.getMessage(), e);
+            throw new RuntimeException(e);        	
         } catch (final Exception e) {
-            LOGGER.error(e.getMessage(), e);
+            LOGGER.error("Error getting response from host: {} with path: {} and protocol: {}: Error Message: {}",
+            		constructedUrl.getHost(), constructedUrl.getPath(), constructedUrl.getProtocol(), e.getMessage(), e);
```

Comment:
I confirmed that a stack trace is printed with the slf4j version the cas client references. I tried passing the non-exception arguments as an Object array and in that case the stack trace was not printed (which I found odd but didn't investigate). Are we concerned about someone using an 8 year old slf4j client with the latest cas client? They would still get exception wrapped as RuntimeException and the previous error message didn't add much in the way of extra information. 

I added a catch and re-throw RuntimeException rather than doing assertNotNull() for the url and factory because this is really an internal only class and we know nobody is passing in nulls so why incur any overhead no matter how negligible? I just don't want anyone trying to pass a null argument to get NPE in my log statement. 


This is output from a temporary main method I added to the class. I don't know how to test log messages from a unit test. 
```
[main] ERROR org.jasig.cas.client.util.CommonUtils - SSL error getting response from host: www.gooogle.com : Error Message: java.security.cert.CertificateException: No subject alternative DNS name matching www.gooogle.com found.
javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: No subject alternative DNS name matching www.gooogle.com found.
	at sun.security.ssl.Alerts.getSSLException(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.fatal(Unknown Source)
	at sun.security.ssl.Handshaker.fatalSE(Unknown Source)
	at sun.security.ssl.Handshaker.fatalSE(Unknown Source)
	at sun.security.ssl.ClientHandshaker.serverCertificate(Unknown Source)
	at sun.security.ssl.ClientHandshaker.processMessage(Unknown Source)
	at sun.security.ssl.Handshaker.processLoop(Unknown Source)
	at sun.security.ssl.Handshaker.process_record(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.readRecord(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.afterConnect(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
	at org.jasig.cas.client.util.CommonUtils.getResponseFromServer(CommonUtils.java:433)
	at org.jasig.cas.client.util.CommonUtils.getResponseFromServer(CommonUtils.java:403)
	at org.jasig.cas.client.util.CommonUtils.main(CommonUtils.java:718)
Caused by: java.security.cert.CertificateException: No subject alternative DNS name matching www.gooogle.com found.
	at sun.security.util.HostnameChecker.matchDNS(Unknown Source)
	at sun.security.util.HostnameChecker.match(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.checkIdentity(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.checkIdentity(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.checkTrusted(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(Unknown Source)
	... 16 more
[main] ERROR org.jasig.cas.client.util.CommonUtils - Error getting response from host: [www.gooogle213.com] with path: [] and protocol: [https] Error Message: www.gooogle213.com
java.net.UnknownHostException: www.gooogle213.com
	at java.net.AbstractPlainSocketImpl.connect(Unknown Source)
	at java.net.PlainSocketImpl.connect(Unknown Source)
	at java.net.SocksSocketImpl.connect(Unknown Source)
	at java.net.Socket.connect(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.connect(Unknown Source)
	at sun.security.ssl.BaseSSLSocketImpl.connect(Unknown Source)
	at sun.net.NetworkClient.doConnect(Unknown Source)
	at sun.net.www.http.HttpClient.openServer(Unknown Source)
	at sun.net.www.http.HttpClient.openServer(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.<init>(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.New(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
	at org.jasig.cas.client.util.CommonUtils.getResponseFromServer(CommonUtils.java:433)
	at org.jasig.cas.client.util.CommonUtils.getResponseFromServer(CommonUtils.java:403)
	at org.jasig.cas.client.util.CommonUtils.main(CommonUtils.java:723)
```